keras_cv/metrics/coco/recall.py

```diff
@@ -12,11 +12,173 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import tensorflow as tf
+import tensorflow.keras as keras
+import tensorflow.keras.initializers as initializers
 
-from keras_cv.metrics.coco.base import COCOBase
+from keras_cv.metrics.coco import iou as iou_lib
+from keras_cv.metrics.coco import utils
+from keras_cv.utils import bbox
 
 
-class COCORecall(COCOBase):
+class COCORecall(keras.metrics.Metric):
+    """COCORecall computes the COCO recall metric.
+
+    Args:
+        iou_thresholds: defaults to [0.5:0.05:0.95].
+        category_ids: no default, users must provide.
+        area_range: area range to consider bounding boxes in. Defaults to all.
+        max_detections: number of maximum detections a model is allowed to make.
+
+    Usage:
+        COCORecall accepts two Tensors as input to it's `update_state` method.
+        These Tensors represent bounding boxes in `corners` format.  Utilities
+        to convert Tensors from `xywh` to `corners` format can be found in
+        `keras_cv.utils.bbox`.
+
+        Each image in a dataset may have a different number of bounding boxes,
+        both in the ground truth dataset and the prediction set.  In order to
+        account for this, you may either pass a `tf.RaggedTensor`, or pad Tensors
+        with `-1`s to indicate unused boxes.  A utility function to perform this
+        padding is available at `keras_cv_.utils.bbox.pad_bbox_batch_to_shape`.
+
+        ```
+        coco_recall = COCORecall(
+            max_detections=100,
+            category_ids=[1],
+            area_range=(32**2, 64**2),
```

Comment:
Might want to add this example to area_range docstring, eg it is expecting a list of 2 ints, which are lower and upper bound of size of the bbox.